### PR TITLE
feat(argo-workflows): Add dev/delegated auth

### LINKED ISSUE #895: [argo-workflows] Argo UI silently defaults to authentication since chart 0.4.0
### Describe the bug

Since argo-workflows chart v0.4.0, the user interface to Argo server requires authentication by default (probably since introduction of a 3.0.7+ version of `appVersion`. I think I traced this to commit https://github.com/argoproj/argo-helm/commit/04cbf217aebe9ce3e847b277ecb4c25eb95fbf2d, before then, I was able to access the web UI without any kind of authentication.

Since our gateway handles the authentication part, I have no need for additional authentication layers. But I haven't been able to find the documentation on how to disable something that was disabled by default before.

The behaviour can be disabled by adding
```yaml
server:
  extraArgs:
    - --auth-mode=server
```
to the configured values. If this is no longer the desired default, this should ideally be documented.

### To Reproduce

1. Install argo-workflows helm chart v0.3.0 
2. Visit http://{your_domain/and/any/subpaths}/argo. You have immediate access to the UI
3. Deploy argo-workflows helm chart v0.4.0
4. Visit http://{your_domain/and/any/subpaths}/argo again, you get a login screen (see below)

### Expected behavior

I'd like access to the web UI without authentication please

### Screenshots

image

### Additional context
Added the pod manifest as deployed locally on my machine
[argo-server.zip](https://github.com/argoproj/argo-helm/files/7039757/argo-server.zip)

### PR DESCRIPTION
**Pull request by reinvantveer**
This PR resolves #895, mostly by documenting the default argo server authentication behaviour of the argo workflows chart. It has commented-out settings that can be either easily enabled or overridden, and it shows how to do so. 

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

Closes #895

### CODE REVIEW

File: charts/argo-workflows/README.md [hunk 1]
```diff
@@ -175,7 +181,7 @@ Fields to note:
 | server.serviceType | string | `"ClusterIP"` | Service type for server pods |
 | server.sso | object | `{}` | SSO configuration when SSO is specified as a server auth mode. |
 | server.tolerations | list | `[]` | [Tolerations] for use with node taints |
-| server.volumeMounts | list | `[]` | Additional volume mounts to the server main container. |
```
Comment (mkilchhofer) [inline charts/argo-workflows/README.md:178]: Hmm. Is this change intentional?

Comment (reinvantveer) [reply charts/argo-workflows/README.md:178]: Wow, don't quite know what happened there, lemme fix

**Comment by reinvantveer** [comment]
Thank you @mkilchhofer, @mbevc1!

**Comment by mbevc1** [comment]
Thanks for your contribution :+1:
